app/javascript/components/TextTruncate.tsx

```diff
@@ -10,21 +10,33 @@ import { renderInlineMarkup } from "../util/languageUtil"
 
 export interface TextTruncateProps {
   text: string
+  className?: string
+  buttonClassName?: string
 }
 
-export const TextTruncate = ({ text }: TextTruncateProps) => {
+export const TextTruncate = ({ text, className, buttonClassName }: TextTruncateProps) => {
   const [isExpanded, setIsExpanded] = useState(false)
 
   truncate.setup({ ellipsis: " ... ", length: 400, reserveLastWord: true, keepWhitespaces: true })
   const truncatedText = text ? renderInlineMarkup(truncate(text)) : ""
   const untruncatedText = renderInlineMarkup(text)
+  let wrapperClassNames = ["text-sm"]
+  if (className) {
+    wrapperClassNames = [...wrapperClassNames, className]
+  }
+
+  let buttonClassNames = ["button-toggle ml-1"]
+
+  if (buttonClassName) {
+    buttonClassNames = [...buttonClassNames, buttonClassName]
+  }
 
   return (
-    <div className="text-sm">
+    <span className={wrapperClassNames.join(" ")}>
       {isExpanded ? untruncatedText : truncatedText}
```

Comment:
is this ok to be a span? any concerns about invalid markup?

Reply:
@christine-sfg Changed it to a div and added an inline display!